api/api/utils/moderation_lock.py

```diff
@@ -0,0 +1,123 @@
+import functools
+import time
+
+import django_redis
+import structlog
+from redis.exceptions import ConnectionError
+
+from api.models.moderation import get_moderators
+
+
+LOCK_PREFIX = "moderation_lock"
+TTL = 10  # seconds
+
+logger = structlog.get_logger(__name__)
+
+
+def handle_redis_exception(func):
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except ConnectionError:
+            return None
+
+    return wrapper
```

Comment:
Nice idea. I wish we could use something like this in other parts of the app that handle Redis connection failures to reduce all those try/except blocks.